storage/xtradb/handler/ha_innodb.cc

```diff
@@ -11025,6 +11064,22 @@ create_table_check_doc_id_col(
 	return(false);
 }
 
+/*********************************************************************//*
+ creates a column name for hash index */
+UNIV_INTERN
+char*
+innobase_create_column_name_for_hash_index(
+	/*=====================================*/
+	mem_heap_t*		heap,	    /*!< in: memory heap */
+	const char*		index_name)	/*!< in: name of index */
+{
+	ulint len = strlen("HASH_COL_");
+	ulint len1 = strlen("index_name");
+	char* col_name = static_cast<char*>(mem_heap_alloc(heap, len + len1 + 1));
+	memcpy(col_name, "HASH_COL_", len);
+	memcpy(col_name + len, index_name, len1 + 1);
```

Comment:
This is then incorrect, len1+1 is 11, but actual index_name length could be longer.